vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,539 @@
+Nowcasting uses the previously observed reporting delays to correct for this downward bias and estimate the eventual, final observed cases, which provides an estimate of the trend in cases in real-time.
+
+## About syndromic surveillance system data
+
+Syndromic surveillance system data contains information at the visit-level on the timing and nature of the patient's clinical interactions.
+In this case study, we will use synthetic data from the United States' [National Syndromic Surveillance Program](https://www.cdc.gov/nssp/index.html) (NSSP) dataset available for participating jurisdictions through the CDC's ESSENCE platform, to nowcast cases of Broad Acute Respiratory Incidence (BAR).
+The NSSP Emergency Department (ED) visit dataset is a dataset widely used by public health departments in the United States, representing many but not all counties in the country.
+The dataset contains information at the visit-level about each clinical encounter recorded in the electronic health record system during the patient's hospital stay.
+Clinical encounters may begin to be associated with diagnoses codes at different points in the patient or clinical processing journey (e.g. during registration, triage, clinical encounter, after laboratory results are returned, or during coding for billing, etc.), and are captured through update messages to the syndromic surveillance system once they are entered into a facility's electronic health record.
+The difference between the visit date - when the patient registers in the emergency department- and the time of the diagnosis update pertaining to the diagnosis of interest, is used to compute a reporting delay for each patient.
+Reporting delays for diagnoses of interest can vary by a range of factors including by pathogen/syndrome, season, time of day or week, means of diagnosis, the electronic health record system, or treating facility.
+In this vignette, we will demonstrate how to use syndromic surveillance system data, particularly the timestamped data captured in each NSSP ESSENCE visit record, to produce nowcasts of cases of a particular diagnosis of interest.
+Note that all visits that originate in the emergency department are used for this analysis, regardless of eventual inpatient admission.
+
+
+## Load packages
+
+We use the `baselinenowcast` package, `dplyr`, and `tidyr` for data manipulation, `readr` for loading in data, `stringr` for parsing text data, `lubridate` for formatting dates, `ggplot2` for plotting, and `purrr` for mapping diagnoses codes to text fields in the data.
+```{r setup, message = FALSE, warning = FALSE, results = 'hide'}
+# Load packages
+library(baselinenowcast)
+library(dplyr)
+library(tidyr)
+library(readr)
+library(stringr)
+library(lubridate)
```

Comment:
are you actually using lubridate now?

Reply:
Yes for `days(delay)`

Reply:
I think you may be able to avoid that if you would lke it